### PR TITLE
FEAT: exeption handling, date bug fixed

### PR DESCRIPTION
**Pull request by ArthurO2002**

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -15,6 +17,21 @@ initializeApp({
...
+    "string.required": "Today's work field is required",
+    "string.min": "Today's work need's at least 1 character",
+  }),
+  yesterdayWork: Joi.string().required().messages({
+    "string.required": "Yesterday's work field is required",
+    "string.min": "Yesterday's work need's at least 1 character",
+  }),
+  date: Joi.date().min(new Date(2010, 1, 1)).required().messages({
+    "date.required": "Date field can't be empty",
+    "date.min": "Date field Should not be empty or older than 2010-01-01",
```
Comment (harutyunyan611) [inline functions/src/index.ts]: Can we have a separate file for messages?

File: functions/src/index.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
 /* eslint-disable camelcase */
+/* eslint-disable max-len */
```
Comment (harutyunyan611) [inline functions/src/index.ts]: Can we avoid using eslint-disable

File: functions/src/index.ts [hunk 3]
```diff
@@ -44,22 +61,29 @@ interface IContainer {
   message_ts: string
   type: string
 }
+interface IActions {
+  action_id: string,
+  action_ts: string,
+  type:string
```
Comment (harutyunyan611) [inline functions/src/index.ts]: Not consistent spacing

File: functions/src/index.ts [hunk 4]
```diff
@@ -121,6 +151,7 @@ export const myBot = functions.https.onRequest( async (req, res) => {
       res.status(200);
     } else {
       console.log("Tokens don't match");
+      return;
```
Comment (harutyunyan611) [inline functions/src/index.ts:159]: Why not forbidden error?

File: functions/src/index.ts [hunk 5]
```diff
@@ -87,23 +110,30 @@ export const myBot = functions.https.onRequest( async (req, res) => {
         todayWork: data.todayWork,
         yesterdayWork: data.yesterdayWork,
       };
-      console.log(finalData);
+      const value = schema.validate(finalData);
+      console.log(JSON.stringify(value));
```
Comment (harutyunyan611) [inline functions/src/index.ts]: Please remove unneeded consoles

File: functions/src/index.ts [hunk 6]
```diff
@@ -44,22 +61,29 @@ interface IContainer {
...
   if (req.body.command === "/make-status-update") {
     const modal = inputModal;
     res.send(modal);
     res.status(200);
-  } else {
-    const body: IBody = JSON.parse(req.body.payload);
-    console.log(body.token, config.slack.verification);
+    return;
+  }
+  const body: IBody = JSON.parse(req.body.payload);
```
Comment (harutyunyan611) [inline functions/src/index.ts]: What will be if payload will be non-parsable?

**Review by harutyunyan611 — CHANGES_REQUESTED** [review]

File: functions/src/index.ts (hunk 1 above)
Comment (ArthurO2002) [reply functions/src/index.ts]: Done

File: functions/src/index.ts (hunk 2 above)
Comment (ArthurO2002) [reply functions/src/index.ts]: Done

File: functions/src/index.ts (hunk 3 above)
Comment (ArthurO2002) [reply functions/src/index.ts]: Done

File: functions/src/index.ts (hunk 5 above)
Comment (ArthurO2002) [reply functions/src/index.ts]: Done

File: functions/src/index.ts (hunk 6 above)
Comment (ArthurO2002) [reply functions/src/index.ts]: Done

File: functions/src/index.ts (hunk 4 above)
Comment (ArthurO2002) [reply functions/src/index.ts:159]: Done

File: functions/src/index.ts [hunk 7]
```diff
@@ -44,22 +52,37 @@ interface IContainer {
...
+    return;
+  }
+  let body: IBody;
+  try {
+    body = JSON.parse(req.body.payload);
+  } catch (err) {
+    console.log(err);
+    res.status(400);
+    res.send("not parsable object");
+    throw new Error("not parsable object");
```
Comment (harutyunyan611) [inline functions/src/index.ts]: Unclear why res rend and error

**Review by harutyunyan611 — CHANGES_REQUESTED** [review]

File: functions/src/index.ts (hunk 7 above)
Comment (ArthurO2002) [reply functions/src/index.ts]: Done

File: functions/src/index.ts (hunk 4 above)
Comment (harutyunyan611) [reply functions/src/index.ts:159]: Access denied, Invalid access token

File: functions/src/index.ts [hunk 8]
```diff
@@ -44,22 +52,37 @@ interface IContainer {
...
-    console.log(body.token, config.slack.verification);
+    return;
+  }
+  let body: IBody;
+  try {
+    body = JSON.parse(req.body.payload);
+  } catch (err) {
+    console.log(err);
+    res.status(400);
+    res.send("not parsable object");
```
Comment (harutyunyan611) [inline functions/src/index.ts]: Unable to parse the payload

Comment (ArthurO2002) [reply functions/src/index.ts]: Done

**Review by harutyunyan611 — APPROVED** [review]

File: functions/src/index.ts (hunk 4 above)
Comment (ArthurO2002) [reply functions/src/index.ts:159]: Done

**Pull request merged**